openapi-cli/src/main/java/io/ballerina/openapi/generators/service/ServiceGenerationUtils.java

```diff
@@ -0,0 +1,217 @@
+                                        getItems().get$ref())));
+                    } else if (!(((ArraySchema) schema).getItems() instanceof ArraySchema)) {
+                        member = createBuiltinSimpleNameReferenceNode(null,
+                                createIdentifierToken("string"));
+                    } else {
+                        member = createBuiltinSimpleNameReferenceNode(null,
+                                createIdentifierToken(convertOpenAPITypeToBallerina(
+                                        ((ArraySchema) schema).getItems().getType())));
+                    }
+                    return  createArrayTypeDescriptorNode(member, createToken(SyntaxKind.OPEN_BRACKET_TOKEN), null,
+                            createToken(SyntaxKind.CLOSE_BRACKET_TOKEN));
+                } else {
+                    identifierToken =  createIdentifierToken(schema.getType() + " ");
+                }
+            } else if (schema instanceof ComposedSchema) {
+                if (((ComposedSchema) schema).getOneOf() != null) {
+                    Iterator<Schema> iterator = ((ComposedSchema) schema).getOneOf().iterator();
+                    return getUnionNodeForOneOf(iterator);
+                } else {
+                    identifierToken =  createIdentifierToken("json ");
+                }
+            } else {
+                identifierToken =  createIdentifierToken("json ");
+            }
+        } else {
+            identifierToken =  createIdentifierToken("json ");
+        }
+        return createSimpleNameReferenceNode(identifierToken);
+    }
+    /**
+     * Generate TypeDescriptor for all the mediaTypes.
+     */
+    public static TypeDescriptorNode getMediaTypeToken(Map.Entry<String, MediaType> mediaType)
+            throws BallerinaOpenApiException {
+        String mediaTypeContent = mediaType.getKey().trim();
+        MediaType value = mediaType.getValue();
+        Schema schema = value.getSchema();
+        IdentifierToken identifierToken;
+        switch (mediaTypeContent) {
+            case "application/json":
+                return getIdentifierTokenForJsonSchema(schema);
+            case "application/xml":
+                identifierToken = createIdentifierToken("xml");
+                return createSimpleNameReferenceNode(identifierToken);
+            case "text/plain":
+                identifierToken = createIdentifierToken("string");
+                return createSimpleNameReferenceNode(identifierToken);
+            case "application/octet-stream":
+                return createArrayTypeDescriptorNode(createBuiltinSimpleNameReferenceNode(
+                        null, createIdentifierToken("byte")), createToken(SyntaxKind.OPEN_BRACKET_TOKEN),
+                        null, createToken(SyntaxKind.CLOSE_BRACKET_TOKEN));
+            default:
+                identifierToken = createIdentifierToken("json");
+                return createSimpleNameReferenceNode(identifierToken);
```

Comment:
Lets use constants/enums for Ballerina built-in types. Please revisit other related places as well